### PR TITLE
feat(ThirdParty): Add USS Library Support

### PR DESCRIPTION
**Pull request by Jake-Carter**
### Description

Adds USS library support to libs.mk (enable via `LIB_USS = 1`).

Additionally adds some modifications to the lwIP stack to support the USS library.

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

File: Libraries/lwIP/include/Maxim/arch/sys_arch.h [hunk 1]
```diff
@@ -1,35 +1,95 @@
-/******************************************************************************
+/*
+ * Copyright (c) 2017 Simon Goldschmidt
+ * All rights reserved.
  *
- * Copyright (C) 2022-2023 Maxim Integrated Products, Inc. (now owned by 
- * Analog Devices, Inc.),
- * Copyright (C) 2023-2024 Analog Devices, Inc.
```
Comment (Jake-Carter) [inline Libraries/lwIP/include/Maxim/arch/sys_arch.h:5]: @demiromerb copyright changed on this file - is this correct?

File: Libraries/lwIP/include/Maxim/arch/sys_arch.h [hunk 2]
```diff
@@ -1,35 +1,95 @@
...
+  void *mut;
+};
+typedef struct _sys_mut sys_mutex_t;
+#define sys_mutex_valid_val(mutex)   ((mutex).mut != NULL)
+#define sys_mutex_valid(mutex)       (((mutex) != NULL) && sys_mutex_valid_val(*(mutex)))
+#define sys_mutex_set_invalid(mutex) ((mutex)->mut = NULL)
+#endif /* !LWIP_COMPAT_MUTEX */
 
-#ifndef LIBRARIES_LWIP_INCLUDE_MAXIM_ARCH_SYS_ARCH_H_
-#define LIBRARIES_LWIP_INCLUDE_MAXIM_ARCH_SYS_ARCH_H_
```
Comment (Jake-Carter) [inline Libraries/lwIP/include/Maxim/arch/sys_arch.h:22]: Need to re-add header guard with this string `LIBRARIES_LWIP_INCLUDE_MAXIM_ARCH_SYS_ARCH_H_`

Comment (demiromerb) [reply Libraries/lwIP/include/Maxim/arch/sys_arch.h:22]: That's OK

File: Libraries/lwIP/include/Maxim/arch/sys_arch.h (hunk 1 above)
Comment (demiromerb) [reply Libraries/lwIP/include/Maxim/arch/sys_arch.h:5]: Yes, that is correct. The old file was just a placeholder for some variables. New file is fetched directly from the copyright holders work. Nothing changed by us so,  as far as I know we cannot add our copyright on top of that.

File: Libraries/libs.mk [hunk 3]
```diff
@@ -319,3 +319,18 @@ LIB_CLI_DIR ?= $(LIBS_DIR)/CLI
...
 # ************************
+
+# Unified Security Software (USS) (Disabled by default)
+# Only available via NDA
+# ************************
+LIB_USS ?= 0
+ifeq ($(LIB_USS),1)
+LIB_USS_DIR ?= $(LIBS_DIR)/USS
+
+ifeq ("$(wildcard $(LIB_USS))","")
```
Comment (demiromerb) [inline Libraries/libs.mk:330]: LIB_USS should be LIB_USS_DIR

File: Libraries/lwIP/Maxim/mxc_ppp.c [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+#include <stdio.h>
+#include <string.h>
+
+#include "lwip/init.h"
+#include "lwip/opt.h"
+#include "arch/cc.h"
+
+#include "lwip/sio.h"
+
+#include "mxc_ppp.h"
```
Comment (demiromerb) [inline Libraries/lwIP/Maxim/mxc_ppp.c:34]: Can we move 
#include "mxc_ppp.h" 
below 
#if PPP_SUPPORT && PPPOS_SUPPORT /* don't build if not configured for use in lwipopts.h */

**Review by alperenguclu — APPROVED** [review]

**Review by sihyung-maxim — APPROVED** [review]

**Pull request merged**